### PR TITLE
check if user is datasubmitter for file diff view

### PR DESCRIPTION
**Pull request by wajiha-oicr**
**Description of changes**

The old vs new value view in the file difference preview should _also_ be accessible to data submitters.
But data submitters are not meant to see the prioritized row view, that is still for dcc members only.

**Type of Change**

- [ ] Bug
- [ ] Styling
- [x] New Feature

**Checklist before requesting review:**

- [ ] Matches design:

  - component sizes, spacing, and styles
  - font size, weight, colour
  - spelling has been double checked

- [ ] New uikit components have Storybook stories
- [ ] Feature is minimally responsive.
- [ ] Add copyrights to new files
- [ ] Manual testing of UI feature.
- [ ] Selenium test is completed and passing.

### CODE REVIEW

File: components/pages/submission-system/program-clinical-submission/FilesNavigator/FileRecordTable.tsx [hunk 1]
```diff
@@ -116,26 +117,29 @@ export default ({
...
+  );
   const theme = useTheme();
   const { records, stats } = file;
   const fields: ClinicalSubmissionEntityFile['records'][0]['fields'] = records.length
     ? records[0].fields
     : [];
   const sortedRecords = orderBy(
     records,
-    !isDccPreview
+    !(isDccMember(permissions) && isPendingApproval)
```
Comment (hlminh2000) [inline components/pages/submission-system/program-clinical-submission/FilesNavigator/FileRecordTable.tsx:135]: just for my own understanding, why isn't this `!isDiffPreview` :) The original code suggests that would be the equivalent eh?

File: components/pages/submission-system/program-clinical-submission/FilesNavigator/FileRecordTable.tsx [hunk 2]
```diff
@@ -116,26 +117,29 @@ export default ({
   file: ClinicalSubmissionEntityFile;
   submissionData?: React.ComponentProps<typeof SubmissionInfoArea>;
 }) => {
+  const { shortName: programShortName } = usePageQuery<{ shortName: string }>();
   const { token, permissions } = useAuthContext();
-  const isDccPreview = React.useMemo(() => isDccMember(permissions) && isPendingApproval, [
-    token,
-    isPendingApproval,
-  ]);
+  const isDiffPreview = React.useMemo(
```
Comment (hlminh2000) [inline components/pages/submission-system/program-clinical-submission/FilesNavigator/FileRecordTable.tsx:122]: thank you for the meaningful rename! :)

File: components/pages/submission-system/program-clinical-submission/FilesNavigator/FileRecordTable.tsx (hunk 1 above)
Comment (wajiha-oicr) [reply components/pages/submission-system/program-clinical-submission/FilesNavigator/FileRecordTable.tsx:135]: ah, the scope of the ticket said that the row re-ordering step doesn't need to happen for data submitters. That's why this (the orderby sorting function) is the only place I kept the original condition

File: components/pages/submission-system/program-clinical-submission/FilesNavigator/FileRecordTable.tsx (hunk 2 above)
Comment (wajiha-oicr) [reply components/pages/submission-system/program-clinical-submission/FilesNavigator/FileRecordTable.tsx:122]: naming variables takes a lot of brainpower 🧠

File: components/pages/submission-system/program-clinical-submission/FilesNavigator/FileRecordTable.tsx (hunk 1 above)
Comment (hlminh2000) [reply components/pages/submission-system/program-clinical-submission/FilesNavigator/FileRecordTable.tsx:135]: I see :) thank you!

**Pull request merged**